xarray/plot/plot.py

```diff
@@ -13,32 +13,49 @@
 
 import numpy as np
 import pandas as pd
+from datetime import datetime
 
 from .utils import _determine_cmap_params, _infer_xy_labels, get_axis
 from .facetgrid import FacetGrid
 from xarray.core.pycompat import basestring
 
 
-# Maybe more appropriate to keep this in .utils
-def _right_dtype(arr, types):
+def _valid_numpy_subdtype(x, numpy_types):
     """
-    Is the numpy array a sub dtype of anything in types?
+    Is any dtype from numpy_types superior to the dtype of x?
     """
-    return any(np.issubdtype(arr.dtype, t) for t in types)
+    # If any of the types given in numpy_types is understood as numpy.generic,
+    # all possible x will be considered valid.  This is probably unwanted.
+    # Warn then.
+    for t in numpy_types:
+        if np.issubdtype(np.generic, t):
+            warnings.warn('{} is treated as numpy.generic.'.format(t),
```

Comment:
This is an internal, private function. Rather than warning let's issue a `ValueError` to ensure it doesn't happen by accident. It would also be safe to just use an `assert` here, since again it should never happen in user code.